Dockerfile

```diff
@@ -1,19 +1,11 @@
-ARG PHP_VERSION=8.1
-
-FROM php:${PHP_VERSION}-cli
+FROM php:8.1-cli
 
+# Install extensions
 RUN apt-get update && \
     apt-get install -y autoconf pkg-config libssl-dev git unzip libzip-dev zlib1g-dev && \
     pecl install mongodb && docker-php-ext-enable mongodb && \
     pecl install xdebug && docker-php-ext-enable xdebug && \
     docker-php-ext-install -j$(nproc) zip
 
-COPY --from=composer:2.6.2 /usr/bin/composer /usr/local/bin/composer
-
-ENV COMPOSER_ALLOW_SUPERUSER=1
-
-WORKDIR /code
-
-COPY ./ ./
-
-CMD ["bash", "-c", "composer install && ./vendor/bin/phpunit --testdox"]
+# Install Composer
+RUN curl https://getcomposer.org/installer | php -- --install-dir=/usr/local/bin --filename=composer
```

Comment:
Getting the composer executable from the composer image it correct with Docker. It provides more security and allows caching.

Reply:
I replaced this line
```bash
COPY --from=composer:2.6.2 /usr/bin/composer /usr/local/bin/composer
```
Because the composer's version was fixed.
Also, we don't need to cache, because developers build this file one time, and the second reason is if they decide to rebuild this file, they definitely want the updates.

Reply:
Please revert. You can set a less restrictive version constraint.

```php
COPY --from=composer:2 /usr/bin/composer /usr/local/bin/composer
```

> Also, we don't need to cache, because developers build this file one time, and the second reason is if they decide to rebuild this file, they definitely want the updates.

This cache is also cross-projects. It's more durable than the getcomposer.org url.